src/components/Layout/index.tsx

```diff
@@ -49,6 +51,36 @@ const Layout = ({ children }: LayoutProps) => {
     };
   }, []);
 
+  useEffect(() => {
+    // Determines if we are at the top of the page
+    // Locks or unlocks page when pulling down to refresh
+    const pullStart = (e: TouchEvent) => {
+      setPullStartPoint(e.targetTouches[0].screenY);
+
+      const html = document.querySelector('html');
+
+      if (window.scrollY === 0 && window.scrollX === 0) {
+        refreshDiv.current?.classList.add('block');
+        refreshDiv.current?.classList.remove('hidden');
+        document.body.style.touchAction = 'none';
+        document.body.style.overscrollBehavior = 'none';
+        if (html) {
+          html.style.overscrollBehaviorY = 'none';
+        }
+      } else {
+        setPullStartPoint(0);
+        refreshDiv.current?.classList.remove('block');
+        refreshDiv.current?.classList.add('hidden');
+      }
+    };
+
+    window.addEventListener('touchstart', pullStart, { passive: false });
+
+    return () => {
+      window.removeEventListener('touchstart', pullStart);
+    };
+  }, []);
```

Comment:
Why can't this just be in the pull to refresh component and you have the component return null when the conditions are not met? That we we don't have to juggle this state between two components?